src/anndata/_io/specs/methods.py

```diff
@@ -603,7 +633,13 @@ def write_recarray_zarr(
 ):
     from anndata.compat import _to_fixed_length_strings
 
-    f.create_dataset(k, data=_to_fixed_length_strings(elem), **dataset_kwargs)
+    elem = _to_fixed_length_strings(elem)
+    if isinstance(f, H5Group) or is_zarr_v2():
+        f.create_dataset(k, data=elem, shape=elem.shape, **dataset_kwargs)
+    else:
+        # TODO: zarr on-disk v3 doesn’t support this dtype
```

Comment:
this seems like a rather large TODO!

Reply:
Ah, so the issue is that I don’t know what that is.

I assumed this means “on-disk as opposed to in-memory”, i.e. all file-based stores and not some specific feature.

Reply:
Right, so there is zarr-python the package, which had a major version upgrade to 3.  There is also the on-disk v3 format, which is different than v2.  v3 does not support recarray.

Reply:
got it! could you rephrase a bit then, please?

```suggestion
        # TODO: zarr’s on-disk format v3 doesn’t support this dtype
```